### PR TITLE
Refactor `isSettled` to allow better insight into current state.

### PR DESCRIPTION
**Pull request by rwjblue**
`isSettled()` will return boolean, and `getSettledState()` will return an object with a number of properties that you can use to see _why_ `isSettled()` returns `false`...

Also:

* Add actual unit tests for `settled`.
* Extract `tests/helpers/ajax.js` to ease testing requests.
* Move existing settled tests to clearly be "integration".

### CODE REVIEW

**Comment by rwjblue** [comment]
For any reviewers, its probably going to be easier to review as separate commits...

**Comment by rwjblue** [comment]
The failures here are from moving the tests around (from `tests/unit` to `tests/integration`). Which means..... we have order dependent tests 😭 

https://github.com/emberjs/ember-test-helpers/pull/252 starts tackling issues (identified via `http://localhost:4200/?seed` and re-run until passing routinely)...

**Comment by rwjblue** [comment]
Rebased to include the changes from #252...

**Comment by Turbo87** [comment]
looks like this fails on Ember 2.4 for some reason 🤔

**Comment by rwjblue** [comment]
Ya, it has to do with a bug in 2.4, but I’m going to have to track it down and guard for it.

**Comment by rwjblue** [comment]
FINALLY
